crates/bevy_ui/src/layout/mod.rs

```diff
@@ -45,22 +45,24 @@ impl LayoutContext {
 #[derive(Debug, Clone, PartialEq, Eq)]
 struct RootNodePair {
     // The implicit "viewport" node created by Bevy
-    implicit_viewport_node: taffy::node::Node,
+    implicit_viewport_node: taffy::tree::NodeId,
     // The root (parentless) node specified by the user
-    user_root_node: taffy::node::Node,
+    user_root_node: taffy::tree::NodeId,
 }
 
+type UiTaffyTree = TaffyTree<Box<dyn Measure>>;
```

Comment:
Now that the measurement/node context type is user-defined, should we use a enum with one variant for each of `FixedMeasure`, `ImageMeasure`... internal to `bevy_ui` plus a `Box<dyn Measure>` variant, to avoid doing dynamic dispatch for most use cases?

Reply:
Yes.